kube-api-test flaky KubernetesValidationHookHandlingTest: Kube API Server did not start properly

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 60.17 s <<< FAILURE! -- in io.fabric8.kubeapitest.sample.KubernetesValidationHookHandlingTest
2026-05-19T06:56:55.5704403Z [ERROR] io.fabric8.kubeapitest.sample.KubernetesValidationHookHandlingTest -- Time elapsed: 60.17 s <<< ERROR!
2026-05-19T06:56:55.5705079Z io.fabric8.kubeapitest.KubeAPITestException: Kube API Server did not start properly
2026-05-19T06:56:55.5705748Z 	at io.fabric8.kubeapitest.process.ProcessReadinessChecker.pollWithTimeout(ProcessReadinessChecker.java:141)
2026-05-19T06:56:55.5706523Z 	at io.fabric8.kubeapitest.process.ProcessReadinessChecker.waitUntilReady(ProcessReadinessChecker.java:129)
2026-05-19T06:56:55.5707268Z 	at io.fabric8.kubeapitest.process.KubeAPIServerProcess.waitUntilReady(KubeAPIServerProcess.java:109)
2026-05-19T06:56:55.5708018Z 	at io.fabric8.kubeapitest.KubeAPIServer.start(KubeAPIServer.java:61)
2026-05-19T06:56:55.5708608Z 	at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.startApiServer(KubeAPIServerExtension.java:95)
2026-05-19T06:56:55.5709472Z 	at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.lambda$startIfAnnotationPresent$2(KubeAPIServerExtension.java:88)
2026-05-19T06:56:55.5710091Z 	at java.base/java.util.Optional.ifPresent(Optional.java:178)
2026-05-19T06:56:55.5710864Z 	at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.lambda$startIfAnnotationPresent$3(KubeAPIServerExtension.java:81)
2026-05-19T06:56:55.5711485Z 	at java.base/java.util.Optional.ifPresent(Optional.java:178)
2026-05-19T06:56:55.5712078Z 	at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.startIfAnnotationPresent(KubeAPIServerExtension.java:78)
2026-05-19T06:56:55.5712822Z 	at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.initialize(KubeAPIServerExtension.java:71)
2026-05-19T06:56:55.5713501Z 	at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.beforeAll(KubeAPIServerExtension.java:48)
2026-05-19T06:56:55.5714031Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
2026-05-19T06:56:55.5714637Z 	Suppressed: io.fabric8.kubeapitest.KubeAPITestException: java.nio.file.NoSuchFileException: /tmp/etcddata1827503843030804988
2026-05-19T06:56:55.5715333Z 		at io.fabric8.kubeapitest.process.EtcdProcess.cleanEtcdData(EtcdProcess.java:107)
2026-05-19T06:56:55.5715834Z 		at io.fabric8.kubeapitest.KubeAPIServer.stop(KubeAPIServer.java:72)
2026-05-19T06:56:55.5716472Z 		at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.lambda$stopIfAnnotationPresent$5(KubeAPIServerExtension.java:103)
2026-05-19T06:56:55.5717088Z 		at java.base/java.util.Optional.ifPresent(Optional.java:178)
2026-05-19T06:56:55.5717691Z 		at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.lambda$stopIfAnnotationPresent$6(KubeAPIServerExtension.java:101)
2026-05-19T06:56:55.5718303Z 		at java.base/java.util.Optional.ifPresent(Optional.java:178)
2026-05-19T06:56:55.5718898Z 		at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.stopIfAnnotationPresent(KubeAPIServerExtension.java:99)
2026-05-19T06:56:55.5719634Z 		at io.fabric8.kubeapitest.junit.KubeAPIServerExtension.afterAll(KubeAPIServerExtension.java:53)
2026-05-19T06:56:55.5720074Z 		... 1 more
2026-05-19T06:56:55.5720361Z 	Caused by: java.nio.file.NoSuchFileException: /tmp/etcddata1827503843030804988
2026-05-19T06:56:55.5720963Z 		at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
2026-05-19T06:56:55.5721498Z 		at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
2026-05-19T06:56:55.5722027Z 		at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
2026-05-19T06:56:55.5722617Z 		at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
2026-05-19T06:56:55.5723256Z 		at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:171)
2026-05-19T06:56:55.5724079Z 		at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
2026-05-19T06:56:55.5724605Z 		at java.base/java.nio.file.Files.readAttributes(Files.java:1854)
2026-05-19T06:56:55.5725125Z 		at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:220)
2026-05-19T06:56:55.5725612Z 		at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:277)
2026-05-19T06:56:55.5726056Z 		at java.base/java.nio.file.FileTreeWalker.walk(FileTreeWalker.java:323)
2026-05-19T06:56:55.5726516Z 		at java.base/java.nio.file.FileTreeIterator.<init>(FileTreeIterator.java:71)
2026-05-19T06:56:55.5726924Z 		at java.base/java.nio.file.Files.walk(Files.java:3911)
2026-05-19T06:56:55.5727248Z 		at java.base/java.nio.file.Files.walk(Files.java:3966)
2026-05-19T06:56:55.5727614Z 		at io.fabric8.kubeapitest.Utils.deleteDirectory(Utils.java:105)
2026-05-19T06:56:55.5728089Z 		at io.fabric8.kubeapitest.process.EtcdProcess.cleanEtcdData(EtcdProcess.java:104)
2026-05-19T06:56:55.5728476Z 		... 8 more
```

Check if it relates to #7807

--- a/doc/kube-api-test.md
+++ b/doc/kube-api-test.md
@@ -112,6 +112,11 @@ annotation, since might not make sense to do it for an individual test case. How
 [`KubeApiServer`](https://github.com/fabric8io/kubernetes-client/blob/main/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServer.java)
 and also configured globally using environment variables, see [KubeAPIServerConfigBuilder](https://github.com/fabric8io/kubernetes-client/blob/main/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServerConfigBuilder.java)
 
+The API Server startup wait defaults to 120 seconds (raised from 60 s in [#7834](https://github.com/fabric8io/kubernetes-client/pull/7836)
+to absorb scheduling pressure on shared CI runners). Override per instance with
+`KubeAPIServerConfigBuilder.withStartupTimeout(int)`, or globally via the
+`KUBE_API_TEST_STARTUP_TIMEOUT` environment variable (milliseconds).
+
 
 ### Updating kube config file
 

--- a/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServer.java
+++ b/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServer.java
@@ -34,6 +34,7 @@ public class KubeAPIServer implements UnexpectedProcessStopHandler {
   private final KubeConfig kubeConfig;
   private final EtcdProcess etcdProcess;
   private final KubeAPIServerProcess kubeApiServerProcess;
+  private volatile boolean stopped = false;
 
   public KubeAPIServer() {
     this(KubeAPIServerConfigBuilder.anAPIServerConfig().build());
@@ -62,7 +63,15 @@ public class KubeAPIServer implements UnexpectedProcessStopHandler {
     logger.debug("API Server ready to use");
   }
 
-  public void stop() {
+  // Guarded so the JUnit afterAll path and the processStopped() callback
+  // (fired on unexpected child-process exit during startup) can both call
+  // stop() without the second invocation crashing in cleanEtcdData() once
+  // the temp dirs have already been removed. See #7834.
+  public synchronized void stop() {
+    if (stopped) {
+      return;
+    }
+    stopped = true;
     logger.debug("Stopping");
     kubeApiServerProcess.stopApiServer();
     etcdProcess.stopEtcd();

--- a/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServer.java
+++ b/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServer.java
@@ -34,7 +34,7 @@ public class KubeAPIServer implements UnexpectedProcessStopHandler {
   private final KubeConfig kubeConfig;
   private final EtcdProcess etcdProcess;
   private final KubeAPIServerProcess kubeApiServerProcess;
-  private volatile boolean stopped = false;
+  private boolean stopped = false;
 
   public KubeAPIServer() {
     this(KubeAPIServerConfigBuilder.anAPIServerConfig().build());
@@ -63,10 +63,12 @@ public class KubeAPIServer implements UnexpectedProcessStopHandler {
     logger.debug("API Server ready to use");
   }
 
-  // Guarded so the JUnit afterAll path and the processStopped() callback
-  // (fired on unexpected child-process exit during startup) can both call
-  // stop() without the second invocation crashing in cleanEtcdData() once
-  // the temp dirs have already been removed. See #7834.
+  // stop() can be invoked twice: from the processStopped() callback (fired
+  // on unexpected child-process exit during startup) and from JUnit's
+  // afterAll. The stopped guard makes the second call a no-op so
+  // cleanEtcdData() doesn't crash on the already-removed temp dirs;
+  // synchronized closes the rarer window where the two callers overlap
+  // concurrently and would otherwise race past the guard. See #7834.
   public synchronized void stop() {
     if (stopped) {
       return;

--- a/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServerConfigBuilder.java
+++ b/junit/kube-api-test/core/src/main/java/io/fabric8/kubeapitest/KubeAPIServerConfigBuilder.java
@@ -68,7 +68,7 @@ public final class KubeAPIServerConfigBuilder {
     this.apiServerVersion = finalConfigValue(this.apiServerVersion, KUBE_API_TEST_API_SERVER_VERSION, null);
     this.waitForEtcdHealthCheckOnStartup = finalConfigValue(this.waitForEtcdHealthCheckOnStartup,
         KUBE_API_TEST_WAIT_FOR_ETCD_HEALTH_CHECK, false);
-    this.startupTimeout = finalConfigValue(this.startupTimeout, KUBE_API_TEST_STARTUP_TIMEOUT, 60_000);
+    this.startupTimeout = finalConfigValue(this.startupTimeout, KUBE_API_TEST_STARTUP_TIMEOUT, 120_000);
 
     return new KubeAPIServerConfig(apiTestDir, apiServerVersion, offlineMode, apiServerFlags,
         updateKubeConfig, waitForEtcdHealthCheckOnStartup, startupTimeout);

--- a/junit/kube-api-test/core/src/test/java/io/fabric8/kubeapitest/KubeAPIServerConfigBuilderTest.java
+++ b/junit/kube-api-test/core/src/test/java/io/fabric8/kubeapitest/KubeAPIServerConfigBuilderTest.java
@@ -1,0 +1,43 @@
+/*
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubeapitest;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class KubeAPIServerConfigBuilderTest {
+
+  // Default raised from 60s to 120s under #7834: the 2.6s idle startup of
+  // kube-apiserver scales ~20× under -T 1C + forkCount=1C CI contention,
+  // landing right on the 60s boundary. 120s gives a 2× buffer; the
+  // KUBE_API_TEST_STARTUP_TIMEOUT env var still lets users tune it.
+  @Test
+  void defaultStartupTimeoutIs120Seconds() {
+    var config = KubeAPIServerConfigBuilder.anAPIServerConfig().build();
+
+    assertThat(config.getStartupTimeout()).isEqualTo(120_000);
+  }
+
+  @Test
+  void explicitStartupTimeoutOverridesDefault() {
+    var config = KubeAPIServerConfigBuilder.anAPIServerConfig()
+        .withStartupTimeout(42_000)
+        .build();
+
+    assertThat(config.getStartupTimeout()).isEqualTo(42_000);
+  }
+}

--- a/junit/kube-api-test/core/src/test/java/io/fabric8/kubeapitest/sample/KubeApiServerTest.java
+++ b/junit/kube-api-test/core/src/test/java/io/fabric8/kubeapitest/sample/KubeApiServerTest.java
@@ -25,6 +25,7 @@ import org.junit.jupiter.api.Test;
 
 import static io.fabric8.kubeapitest.sample.TestCaseUtils.simpleTest;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 class KubeApiServerTest {
@@ -87,6 +88,21 @@ class KubeApiServerTest {
         .withStartupTimeout(500)
         .build());
     assertThrows(KubeAPITestException.class, kubeApi::start);
+  }
+
+  // Regression for #7834: when an unexpected child-process exit fires the
+  // processStopped() callback during startup, KubeAPIServer.stop() runs from
+  // the callback AND again from JUnit's afterAll. The second invocation used
+  // to crash in EtcdProcess.cleanEtcdData() with NoSuchFileException because
+  // the first call already deleted the temp dirs. stop() must therefore be
+  // idempotent.
+  @Test
+  void stopCanBeCalledMultipleTimes() {
+    var kubeApi = new KubeAPIServer();
+    kubeApi.start();
+    kubeApi.stop();
+
+    assertThatNoException().isThrownBy(kubeApi::stop);
   }
 
   void testWithAPIServer(KubeAPIServer kubeApi) {

